src/Command/SingleDirectoryComponent.php

```diff
@@ -0,0 +1,218 @@
+<?php declare(strict_types = 1);
+
+namespace DrupalCodeGenerator\Command;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Extension\ThemeHandlerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Asset\LibraryDiscovery;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Extension\ExtensionLifecycle;
+use DrupalCodeGenerator\Application;
+use DrupalCodeGenerator\Asset\AssetCollection;
+use DrupalCodeGenerator\Asset\File;
+use DrupalCodeGenerator\Attribute\Generator;
+use DrupalCodeGenerator\GeneratorType;
+use DrupalCodeGenerator\InputOutput\Interviewer;
+use DrupalCodeGenerator\Utils;
+use DrupalCodeGenerator\Validator\Choice;
+use DrupalCodeGenerator\Validator\Optional;
+use DrupalCodeGenerator\Validator\Required;
+use DrupalCodeGenerator\Validator\RequiredMachineName;
+use Symfony\Component\Console\Question\Question;
+
+#[Generator(
+  name: 'single-directory-component',
+  description: 'Generates Drupal SDC theme component',
+  aliases: ['sdc'],
+  templatePath: Application::TEMPLATE_PATH . '/_sdc',
+  type: GeneratorType::THEME_COMPONENT,
+)]
+final class SingleDirectoryComponent extends BaseGenerator implements ContainerInjectionInterface {
+  private const COMPONENT_PATH_TOKEN = '{directory}/{component_machine_name}/';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    private readonly ModuleHandlerInterface $moduleHandler,
+    private readonly ThemeHandlerInterface $themeHandler,
+    private readonly LibraryDiscovery $libraryDiscovery,
+  ) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new self(
+      $container->get('module_handler'),
+      $container->get('theme_handler'),
+      $container->get('library.discovery'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generate(array &$vars, AssetCollection $assets): void {
+    $this->askQuestions($vars);
+    $this->generateAssets($vars, $assets);
+  }
+
+  private function askQuestions(array &$vars): array {
+    $ir = $this->createInterviewer($vars);
+    $vars['machine_name'] = $ir->askMachineName();
+    $vars['name'] = $ir->askName();
+    $vars['directory'] = $ir->ask('Components directory', 'components');
```

Comment:
I wonder if SDC will actually check directories other than 'components'? Event if it does, seems like 'components' directory is established convention. Maybe we don't need to ask this question?

Reply:
I'm validating with e0ipso. I'll get back to you asap.

Reply:
I generally follow the following rule. If 99% users accept default answer that means that the question is likely redundant. Also in this case is not a big deal for users to rename the directory afterwards.